core/modules/field/modules/number/number.module

```diff
@@ -154,6 +154,14 @@ function number_field_validate($entity_type, $entity, $field, $instance, $langco
  * Implements hook_field_presave().
  */
 function number_field_presave($entity_type, $entity, $field, $instance, $langcode, &$items) {
+  // Until https://github.com/backdrop/backdrop-issues/issues/3873
+  // is fully addressed.
+  // Workaround to get number fields with empty values working.
+  if (empty($items)) {
+    $field_name = $field['field_name'];
+    $entity->$field_name = array();
```

Comment:
I think the syntax for this needs to be `$entity->{$field_name} = array();`

Reply:
Not necessarily as it's just a variable, but using the curly brace syntax could reduce the code here.